overlord/ifacestate/ifacemgr.go

```diff
@@ -275,9 +275,23 @@ func (m *InterfaceManager) doRemoveSnapSecurity(task *state.Task, _ *tomb.Tomb)
 	if err != nil {
 		return err
 	}
-	// TODO: remove all connections from the state
-	if err := m.repo.RemoveSnap(snapName); err != nil {
 
+	conns, err := getConns(task.State())
```

Comment:
I see now how there can be multiple snaps with the same name but I'm not sure this correctly translates that to security. Security identifiers don't contain snap revision so removing security for a given snap (at any revision) will impact the security for other snaps with the same name (but with different revision)

I'm not sure how security-affecting tasks and snap-affecting tasks are related wrt multiple revisions now.